Build con GitHub Actions

mnuccioarpae opened the pull request:
@edigiacomo @brancomat Travis+docker difficilmente riesce a finire un build a causa delle nuove limitazioni. Allora ho provato a farlo con GitHub Actions. Per ora mi sono limitato a fare il build su fedora:32 e ha funzionato. Ho usato `.travis-build.sh` senza modificarlo.

Se non ci sono pareri contrari, provo a usarlo per attivare il build su copr.

mnuccioarpae committed:
Prova github actions

mnuccioarpae committed:
GitHub non fornisce centos

mnuccioarpae committed:
Prova build SRPM su fedora

mnuccioarpae committed:
Lancia .travis-build.sh in fedora:32

mnuccioarpae committed:
Segui push anche su branch ghci

Il workflow super-linter veniva eseguito comunque. Questo, invece, non viene eseguito. Proviamo a monitorare anche il branch ghci.

mnuccioarpae committed:
GH Actions richiede .travis-build.sh eseguibile

mnuccioarpae committed:
.travis-build.sh richiede un parametro

mnuccioarpae committed:
Fix date warnings in changelog

mnuccioarpae committed:
Fix date in changelog

mnuccioarpae committed:
Usa git per fare il checkout

actions/checkout usa l'API REST di github per scaricare i sorgenti quando manca git. Però .travis-build.sh usa git per creare l'archivio tar dei sorgenti.

edigiacomo commented:
Come soluzione temporanea direi che va bene, non capisco però se usando Gihub Actions si bypassano i limiti di docker.

mnuccioarpae commented:
@edigiacomo Non uso direttamente docker, quindi, forse, sì. Ho cercato nella documentazione un'affermazione esplicita in merito, ma non l'ho trovata.

edigiacomo commented:
Fa una docker pull come in Travis (https://github.com/ARPA-SIMC/meteo-vm2/pull/23/checks?check_run_id=2181850434#step:2:15), io ho trovato solo questa issue https://github.com/actions/virtual-environments/issues/1445#issuecomment-713861495 ma non è chiarissimo com'è andata a finire.

edigiacomo commented:
Comunque, vai pure con il merge, così lo scopriamo :)

mnuccioarpae committed:
Merge branch 'ghci' of github.com:ARPA-SIMC/meteo-vm2 into ghci

mnuccioarpae commented:
Per provare il collegamento con copr, ho bisogno di poter [salvare il token copr nel repository](https://docs.github.com/en/actions/reference/encrypted-secrets). Io, come semplice membro non posso farlo. @edigiacomo lo puoi fare tu?

mnuccioarpae commented:
@edigiacomo Per intenderci, sto cercando di fare quanto descritto qui: https://pavel.raiskup.cz/blog/github-push-actions-and-copr.html
Con la differenza che noi facciamo l'upload solo del file spec, non dell'srpm (se ho capito bene).

edigiacomo commented:
Ho messo il token nella variabile `COPR_API_TOKEN`

edigiacomo commented:
Scusa, avevo messo un vecchio token, ora è aggiornato.

mnuccioarpae committed:
Build su copr da GH Actions

Proviamo a fare il build con centos:8 invece di fedora:32 (perché questo
è quello che usiamo sui server). Se funziona, inneschiamo il build degli
rpm su copr... forse...
